### PR TITLE
Update webkit-build-archive

### PR DESCRIPTION
**Pull request by commitay**
This Cask has apparently been broken for the last couple of months, it seems this isn't an `app` now, it's a build archive that launches `Safari.app`

webkit

> https://webkit.org/blog/7978/introducing-webkit-build-archives/

> Today we’re excited to roll out some improvements to how we deliver WebKit builds

> With these enhancements, WebKit Build Archives will replace WebKit Nightly builds. 

> The new build archives are intended for engineers that contribute to WebKit development to test and diagnose issues. These builds do not have a signed app launcher nor an auto-update system, making them ill-suited as a daily web browsing application.

> To regularly try out recent WebKit builds on macOS, we recommend downloading and installing Safari Technology Preview, a quality-checked release that automatically updates every two weeks.

### CODE REVIEW

**Comment by commitay** [comment]
```
Cask webkit-nightly depends on macOS release >= 10.13, but you are running release 10.12.
```

File: Casks/webkit-build-archive.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
+cask 'webkit-build-archive' do
+  version :latest
+  sha256 :no_check
+
+  url do
+    require 'open-uri'
+    base_url = 'https://webkit.org/build-archives/'
+    macos_release = if MacOS.version == :sierra
+                      %r{href="([^"]+mac\-sierra[^"]+.zip)"}
+                    else
```
Comment (vitorgalvao) [inline Casks/webkit-build-archive.rb]: I’d rather make this an `elsif` for High Sierra, and leaving the final `else` as an abort or something else, so we’re already safe in six months when the new macOS starts in beta.

Comment (commitay) [reply Casks/webkit-build-archive.rb]: Disabling > `:high_sierra` might give a better error?

```
  depends_on macos: [
                      :sierra,
                      :high_sierra,
                    ]
```

Comment (vitorgalvao) [reply Casks/webkit-build-archive.rb]: @commitay I’m fine with that solution.

**Pull request merged**